Entity/Components/HealthComponent.gd

```diff
@@ -7,32 +7,30 @@ class_name HealthComponent
 ## Emit when health is changed
 signal on_health_changed(new_health: int)
 
+signal on_block_changed(new_block: int)
+
 ## The maximum health the entity can have; this is a hard limit
 @export var max_health: float = 100
 ## The health the entity currently has [br]
 ## Between 0 and max_health
 var current_health: float = 100 # ? change this to max health, or just unset since it's init by the ready
 
+
+var current_block: float = 0
```

Comment:
this should be an `int` not a `float`
and actually all health should be `int`, not `float` but we never changed it since the start

Reply:
Ye, i noticed that, but decided to just make most of the variables floats to keep it consistent. Should I just change all of them to ints?

Reply:
if it doesn't break everything, then yes